builds/ci/dotnet.yaml

```diff
@@ -14,7 +14,7 @@ jobs:
     displayName: Linux
     timeoutInMinutes: 120
     pool:
-      vmImage: ubuntu-16.04
+      vmImage: ubuntu-18.04
```

Comment:
On Line35, we don't need `../templates/install-dotnet3.yaml`. (May be we can also get rid of the `../templates/install-dotnet2.yaml`?! ) 
Please investigate if we can dotnet2 and (hopefully remove it from the rest of the code base).

Reply:
Again, that's not the focus of this PR.